cpp/src/arrow/flight/sql/odbc/flight_sql/config/configuration.cc

```diff
@@ -155,7 +153,14 @@ const std::string& Configuration::Get(const std::string_view& key) const {
 void Configuration::Set(const std::string_view& key, const std::string& value) {
   const std::string copy = boost::trim_copy(value);
   if (!copy.empty()) {
-    this->properties[key] = value;
+    this->properties[std::string(key)] = value;
+  }
+}
+
+void Configuration::Emplace(const std::string_view& key, std::string&& value) {
+  const std::string copy = boost::trim_copy(value);
+  if (!copy.empty()) {
+    this->properties[std::string(key)] = value;
```

Comment:
Use properties.emplace(std::move(copy), value))

Reply:
I misunderstood usage of emplace before, I have fixed this now. In SQLConnect, uid and pwd specified in DSN will take precedence